components/email_aliases/email_aliases_service.cc

```diff
@@ -30,21 +79,125 @@ void EmailAliasesService::BindInterface(
   receivers_.Add(this, std::move(receiver));
 }
 
+void EmailAliasesService::NotifyObserversAuthStateChanged(
+    mojom::AuthenticationStatus status) {
+  for (auto& observer : observers_) {
+    observer->OnAuthStateChanged(
+        mojom::AuthState::New(status, auth_email_, std::nullopt));
+  }
+}
+
 void EmailAliasesService::RequestAuthentication(
     const std::string& auth_email,
     RequestAuthenticationCallback callback) {
-  // TODO: Implement authentication logic
-  std::move(callback).Run("Not implemented");
+  auth_email_ = auth_email;
+  if (auth_email.empty()) {
+    std::move(callback).Run(
+        l10n_util::GetStringUTF8(IDS_EMAIL_ALIASES_ERROR_NO_EMAIL_PROVIDED));
+    return;
+  }
+  AuthenticationRequest auth_request;
+  auth_request.email = auth_email;
+  auth_request.intent = "auth_token";
+  auth_request.service = "email-aliases";
+  const auto body_value = auth_request.ToValue();
+  std::string body = base::WriteJson(body_value).value_or("");
```

Comment:
```cxx
std::string body = base::WriteJson(body_value).value_or("");
```

This silently breaks the request... I wonder if we should have something like
```
std::optional<std::string> body = base::WriteJson(auth_request.ToValue());
CHECK(body)
```

What do you think? Would that be too drastic?